core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockletDataMap.java

```diff
@@ -604,18 +731,26 @@ private ExtendedBlocklet createBlocklet(DataMapRow row, int blockletId) {
     detailInfo.setBlockletId((short) blockletId);
     detailInfo.setDimLens(columnCardinality);
     detailInfo.setSchemaUpdatedTimeStamp(row.getLong(SCHEMA_UPADATED_TIME_INDEX));
-    BlockletInfo blockletInfo = new BlockletInfo();
-    try {
-      byte[] byteArray = row.getByteArray(BLOCK_INFO_INDEX);
-      ByteArrayInputStream stream = new ByteArrayInputStream(byteArray);
-      DataInputStream inputStream = new DataInputStream(stream);
-      blockletInfo.readFields(inputStream);
-      inputStream.close();
-    } catch (IOException e) {
-      throw new RuntimeException(e);
+    byte[] byteArray = row.getByteArray(BLOCK_INFO_INDEX);
+    BlockletInfo blockletInfo = null;
+    if (byteArray.length > 0) {
+      try {
+        blockletInfo = new BlockletInfo();
+        ByteArrayInputStream stream = new ByteArrayInputStream(byteArray);
+        DataInputStream inputStream = new DataInputStream(stream);
+        blockletInfo.readFields(inputStream);
+        inputStream.close();
+        blocklet.setLocation(
```

Comment:
I think this should be outside if block because even if blocklet info does not exist in case of old store still we need to set the file location

Reply:
ok